Sources/Sentry/SentryExtraContextProvider.h

```diff
@@ -0,0 +1,23 @@
+#import "SentryCrashWrapper.h"
+#import "SentryNSProcessInfoWrapper.h"
+#import "SentryUIDeviceWrapper.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Provider of dynamic context data that we need to read at the time of an exception.
+ */
+@interface SentryExtraContextProvider : NSObject
+
++ (instancetype)sharedInstance;
+
+- (_Nonnull instancetype)initWithCrashWrapper:(SentryCrashWrapper *)crashWrapper
```

Comment:
l: nonnull is already assumed because of line 6.

```suggestion
- (instancetype)initWithCrashWrapper:(SentryCrashWrapper *)crashWrapper
```